app/content/plan/index.js

```diff
@@ -1,14 +1,205 @@
-import { PaperProvider, Text } from "react-native-paper";
+import React, { useState } from 'react';
 import { SafeAreaView } from "react-native-safe-area-context";
+import { PaperProvider, Text, Avatar, Button, Card, Appbar } from "react-native-paper";
+import { TouchableOpacity } from 'react-native';
+import { FlatList, StyleSheet, ScrollView, RefreshControl } from 'react-native';
+import Icon from "react-native-vector-icons/MaterialCommunityIcons";
+import { View } from 'react-native';
+import { useNavigation, router } from 'expo-router';
+import { Link } from 'expo-router';
+import drillsData from "~/drill_data.json";
+import { collection, doc, getDoc, getDocs, query, updateDoc, where } from "firebase/firestore";
+import { db } from "~/firebaseConfig";
+import { useEffect } from 'react';
+
+import { formatDate, getUnique } from '~/Utility'
+import { useDrillInfo } from "~/hooks/useDrillInfo";
+import { useUserInfo } from "~/hooks/useUserInfo";
+import Loading from "~/components/loading";
+import { currentAuthContext } from "~/context/Auth";
+
+import { useQueryClient } from '@tanstack/react-query';
+
+
+const DrillList = () => {
+  const queryClient = useQueryClient();
+  const { currentUserId, currentTeamId } = currentAuthContext();
+  const userId = currentUserId ?? null;
+  const teamId = currentTeamId ?? null;
+
+  console.log("USER ID IN PLAN", userId)
+
+  const {
+    data: drillInfo,
+    error: drillInfoError,
+    isLoading: drillInfoIsLoading,
+  } = useDrillInfo();
+
+  const {
+    data: userData,
+    userError: userError,
+    userIsLoading: userIsLoading,
+  } = useUserInfo(userId);
+
+  const [assigned_data, setAssignedData] = useState([]);
+
+  const [refreshing, setRefreshing] = React.useState(false);
+
+  const onRefresh = React.useCallback(() => {
+    const refresh = async () => {
+      setRefreshing(true);
+      await queryClient.invalidateQueries(["user", { teamId, userId }]);
+      setRefreshing(false);
+    };
+
+    refresh();
+  }, [teamId, userId, queryClient]);
+
+
+  console.log("USER DATA", userData);
+
+  // Set the assigned_data state when the user data is loaded
+  useEffect(() => {
+    if (!userIsLoading && userData && userData.assigned_data) {
+      setAssignedData(userData.assigned_data);
+    }
+  }, [userIsLoading, userData]);
+
+  // Group the assigned drills by date
+  const groupedData = assigned_data.reduce((acc, curr) => {
+    const date = convertTimestampToDate(curr.assigned_time);
```

Comment:
Do the "Today" check here and then useMemo to cache it. sth like date = curr.assigned_time === Date.now()/1000 ? "Today" : convertTimestampToDate(curr.assigned_time)

Reply:
and if you're afraid that sortedDates would be hard to implement, just sort assignedData before reducing it.
assignedData.sort().reduce(). This way you don't need sortedDates